Assets/Scripts/SimplePhysics.cs

```diff
@@ -128,28 +136,31 @@ void Update()
             deltaX = GetXChange(deltaX, dirX);
             deltaY = GetYChange(deltaY, dirY, facingRight);
 
-            transform.Translate(deltaX, deltaY, 0f);
-            mVelY += mGravity;
-        }
-
-        private void HandleInput(ref float xInput, ref float yInput, ref bool jumpPressed)
-        {
-            // left/right input
-            if (Mathf.Abs(xInput = Input.GetAxis("LeftStickXAxis")) <= Mathf.Epsilon)
+            // if we have stopped moving in the Y dir,
+            // update our y speed to zero
+            if (deltaY == 0f)
             {
-                xInput = Input.GetAxis("LeftRightKeyboard");
+                mVelY = 0;
             }
 
-            // down input
-            // Note: Pressing down results in positive values so I flip the input
-            if (Mathf.Abs(yInput = -Input.GetAxis("LeftStickYAxis")) <= Mathf.Epsilon)
+            transform.Translate(deltaX, deltaY, 0f);
+            if (mPressedDown)
             {
-                yInput = Input.GetAxis("DownUpKeyboard");
+                mVelY += mGravity * .5f;
             }
-            mPressedDown = Mathf.Sign(yInput) == -1f;
+            mVelY += mGravity;
+        }
 
-            // jump input
-            jumpPressed = Input.GetButtonDown("A") || Input.GetKeyDown(KeyCode.Space);
+        private void HandleInput(ref float xInput, ref float yInput, ref bool jumpPressed)
+        {
+            xInput = InputWrapper.Instance.LeftXInput;
+            yInput = InputWrapper.Instance.LeftYInput;
+            mPressedDown = InputWrapper.Instance.FallPressed;
+            mJumpButtonHeld = InputWrapper.Instance.JumpPressed;
+
+            // disables jump after pressing it
+            jumpPressed = mJumpButtonHeld && mJumpable;
+            mJumpable = (jumpPressed ^ mJumpable) && mJumpable;

```

Comment:
Since `jumpPressed` is only true if `mJumpable` is true, we can just have `mJumpable = mJumpable ^ jumpPressed`